### PR TITLE
Fixed mapping of wxFontStyle to QStyle for fonts

### PR DESCRIPTION
**Pull request by evileye-uk**
Small change to add mapping of some other font styles between Wx and Qt. Fixes a test case.

### CODE REVIEW

File: src/qt/font.cpp [hunk 1]
```diff
@@ -469,9 +469,20 @@ void wxNativeFontInfo::SetPixelSize(const wxSize& size)
 
 void wxNativeFontInfo::SetStyle(wxFontStyle style)
 {
-    m_qtFont.setItalic(style == wxFONTSTYLE_ITALIC);
-//case wxFONTSTYLE_SLANT:
-//case wxFONTSTYLE_NORMAL:
+    switch(style)
+    {
+        case( wxFONTSTYLE_ITALIC ):
```
Comment (vadz) [inline src/qt/font.cpp]: Could you please remove the parentheses around the case labels? This style is not used anywhere else in wx AFAIK.

File: src/qt/font.cpp [hunk 2]
```diff
@@ -469,9 +469,20 @@ void wxNativeFontInfo::SetPixelSize(const wxSize& size)
...
+            m_qtFont.setItalic(true);
+            break;
+        case( wxFONTSTYLE_NORMAL ):
+            m_qtFont.setStyle(QFont::StyleNormal);
+            break;
+        case( wxFONTSTYLE_SLANT ):
+            m_qtFont.setStyle(QFont::StyleOblique);
+            break;
+        default:
```
Comment (vadz) [inline src/qt/font.cpp]: This seems unnecessary and will prevent compiler from warning if another `wxFONTSTYLE_XXX` is added later, so it should really be removed.

File: src/qt/font.cpp [hunk 3]
```diff
@@ -469,9 +469,20 @@ void wxNativeFontInfo::SetPixelSize(const wxSize& size)
 
 void wxNativeFontInfo::SetStyle(wxFontStyle style)
 {
-    m_qtFont.setItalic(style == wxFONTSTYLE_ITALIC);
-//case wxFONTSTYLE_SLANT:
-//case wxFONTSTYLE_NORMAL:
+    switch(style)
+    {
+        case( wxFONTSTYLE_ITALIC ):
+            m_qtFont.setItalic(true);
```
Comment (vadz) [inline src/qt/font.cpp]: Is there any difference between doing this and calling `setStyle(QFont::StyleItalic)`? If not, maybe we should prefer the latter, for consistency?

**Review by vadz — COMMENTED** [review]
This is the right thing to do, of course, thanks for adding support for `wxFONTSTYLE_SLANT`, but could you please address a couple of minor things I've commented on? TIA!

File: src/qt/font.cpp (hunk 3 above)
Comment (liamtreacy) [reply src/qt/font.cpp]: I agree. There's no difference between the two, <setItalic> will ultimately call on <setStyle> so yes, better to have consistency.

**Pull request closed without merging**